tests/integration/gcp.py

```diff
@@ -104,47 +99,27 @@ def _gcp_create_bucket(self, bucket_name):
 
 
     def _gcp_wait_for_operation(self, operation):
-        self.logger.info(f"Waiting for operation {operation['name']} to complete...")
-        kwargs = {"project": self.project, "operation": operation['name']}
-        if 'zone' in operation:
-            client = self._compute.zoneOperations()
-            kwargs["zone"] = operation['zone'].rsplit("/", maxsplit=1)[1]
-        elif 'region' in operation:
-            client = self._compute.regionOperations()
-            kwargs["region"] = operation['region'].rsplit("/", maxsplit=1)[1]
-        else:
-            client = self._compute.globalOperations()
-        response = client.wait(**kwargs).execute()
-
-        if response["status"] != "DONE":
-            self.logger.error("Operation failed %s" % json.dumps(response, indent=4))
-            error = ""
-            if "error" in response:
-                error = response["error"]
-            raise Exception("Operation %s failed: %s" % (operation, error))
+        while not operation.done():
```

Comment:
We should probably use `operation.result()` here instead of `operation.done()`:
* https://googleapis.dev/python/google-api-core/latest/operation.html
* https://googleapis.dev/python/google-api-core/latest/futures.html#google.api_core.future.polling.PollingFuture.result
* https://github.com/googleapis/python-compute/blob/HEAD/samples/snippets/firewall/delete.py